### ISSUE TITLE
Search Rekor for an SBOM attestation

### ISSUE THREAD

**Issue by developer-guy** [anchore/grype#1040]
**What would you like to be added**:

[Rekor](https://github.com/sigstore/rekor) is one of the good public instances served by Sigstore and provides an immutable transparency log for signed metadata related to software supply chain materials such as SBOM, attestation, signature, etc. [cosign](https://github.com/sigstore/cosign) has a sub-command named `attest` that can help you to store an SBOMs in the form of signed attestation and upload it into Rekor.

So, grype can use that SBOM stored in the Rekor to analyze vulnerabilities for the given image.

**Why is this needed**:

Another option for discovering the SBOM information

**Additional context**:
https://blog.aquasec.com/trivy-scans-unpackaged-binary-files

/cc @dentrax @dlorenc @luhring

**Comment by luhring** [comment]
Clarifying question: Is this issue about Grype finding SBOMs for **A)** the image itself, or **B)** files within the image?

**Comment by tgerla** [comment]
Hi @developer-guy, thanks for the suggestion. We're interested in integration here where it makes sense, and after some internal discussion we realized we could use some more details from you as to how this would work. Do you want to stop by our next community meeting (January 19, 12PM Eastern Time) to talk about it? We could also chat on our community Slack. Thanks!

**Comment by developer-guy** [comment]
Yeah, of course, but before then, I need to check how Trivy does to find an SBOM in Rekor. We also discuss this topic in the #ko-build channel in Kubernetes slack. Here are the resources that might help us:

* https://twitter.com/developerguyba/status/1611026853325594624?s=20
* https://kubernetes.slack.com/archives/C01T7DTP65S/p1672933763665089
* https://aquasecurity.github.io/trivy/v0.36/docs/attestation/rekor/

/cc @imjasonh

**Comment by spiffcs** [comment]
If we were to take this on I think this would be implemented as a separate source. We want to be explicit so the user is choosing to pull these from rekor rather than opting out of doing the real source analysis

`grype rekor:alpine:latest` <--- something like
